libp2p/src/main/kotlin/io/libp2p/pubsub/gossip/GossipRouter.kt

```diff
@@ -572,9 +590,14 @@ open class GossipRouter(
         enqueueIwant(peer, messageIds)
     }
 
+    private fun iDontWant(peer: PeerHandler, messageIds: List<MessageId>) {
+        if (messageIds.isEmpty()) return
+        enqueueIdontwant(peer, messageIds)
```

Comment:
As I understand you are going to add the filtering here yet?

Reply:
Yes